src/components/SearchMap.js

```diff
@@ -1,39 +1,62 @@
-import React, {Component} from 'react';
-import {View, ActivityIndicator, StyleSheet, Platform, Keyboard} from 'react-native';
+// 3rd party libraries - core
+import React from 'react';
+import {View, Text, StyleSheet, ActivityIndicator} from 'react-native';
+import {MapView} from 'expo';
 import {Button} from 'react-native-elements';
+// 3rd party libraries - additional
 
-import {MapView} from 'expo';
+// styles and language
+
+// our components - core
+// our components - additional
 
+const SearchMap = ({mapLoaded, region, updateRegion, sites}) => {
+    const {fillScreen, spinnerContainerStyle} = styles;
 
-class SearchMap extends Component {
-    state = {
-        region: {
-            longitude: -105.727939,
-            latitude: 39.695168,
-            longitudeDelta: 1,
-            latitudeDelta: 1
-        },
-        mapLoaded: false
+    const newRegionIsAcceptable = (newRegion) => {
+        const {longitude, latitude, latitudeDelta, longitudeDelta} = newRegion;
+
+        const notZoomedTooFarOut = latitudeDelta < 60 && longitudeDelta < 60;
+        const isNotSF = (Math.abs(longitude - 122.409) > 1) && (Math.abs(latitude - 37.787) > 1);
+
+        return isNotSF && notZoomedTooFarOut;
     };
 
-    componentDidMount() {
-        Keyboard.dismiss();
+    const onRegionChange = (newRegion) => {
 
-        this.setState({
-            mapLoaded: true
-        });
-    }
+        if (!newRegionIsAcceptable(newRegion)) {
+            return;
+        }
 
-    render() {
-        const {fillScreen} = styles;
+        updateRegion(newRegion);
+    };
 
-        return (
-            <View style={fillScreen}>
-                {this.renderMap()}
-            </View>
-        );
-    }
-}
+    const renderMap = () => {
+        if (mapLoaded) {
+            return (
+                <MapView
+                    style={fillScreen}
+                    region={region}
+                    onRegionChange={onRegionChange}
+                    rotateEnabled={false}
```

Comment:
Turned off rotation in the map as this was causing issues. @amgoodman5 let me know if you're passionate about keeping rotation (Is there a good camping reason for this? You'd know better than me), and we can work on keeping it in and debugging performance.

Reply:
Also, we're now working off of `onRegionChange` instead of `onRegionChangeEnabled` since using the enabled one was causing too much jank in zooming in/out of the map.

Reply:
I am not attached to the rotation at all.